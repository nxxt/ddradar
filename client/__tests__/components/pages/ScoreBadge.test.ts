import { createLocalVue, mount } from '@vue/test-utils'
import Buefy from 'buefy'

import ScoreBadge from '~/components/pages/ScoreBadge.vue'
import { ClearLamp } from '~/types/api/score'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('/components/pages/ScoreBadge.vue', () => {
  test.each([
    [7, 1000000] as const,
    [6, 999800] as const,
    [5, 938400] as const,
    [4, 877800] as const,
    [3, 923400] as const,
    [2, 690000] as const,
    [1, 450000] as const,
    [0, 32340] as const,
  ])(
    'renders correctly if props are { lamp: %i, score: %i }',
    (lamp: ClearLamp, score: number) => {
      // Arrange
      const wrapper = mount(ScoreBadge, {
        localVue,
        propsData: { lamp, score },
      })

      // Act - Assert
      expect(wrapper).toMatchSnapshot()
    }
  )
})
