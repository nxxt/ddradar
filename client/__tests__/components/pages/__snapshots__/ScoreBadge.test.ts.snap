// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 0, score: 32340 } 1`] = `
<span data-label="Failed" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-light">
    32340
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 1, score: 450000 } 1`] = `
<span data-label="Assisted Clear" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-challenge">
    450000
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 2, score: 690000 } 1`] = `
<span data-label="Clear" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light">
    690000
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 3, score: 923400 } 1`] = `
<span data-label="Life 4" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-difficult">
    923400
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 4, score: 877800 } 1`] = `
<span data-label="Full Combo" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-beginner">
    877800
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 5, score: 938400 } 1`] = `
<span data-label="Great Full Combo" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-expert">
    938400
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 6, score: 999800 } 1`] = `
<span data-label="Perfect Full Combo" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-basic">
    999800
  </span></span>
`;

exports[`/components/pages/ScoreBadge.vue renders correctly if props are { lamp: 7, score: 1000000 } 1`] = `
<span data-label="Marvelous Full Combo" class="is-primary is-top is-medium b-tooltip" style="transition-delay: 0ms;"><span class="tag is-light is-white">
    1000000
  </span></span>
`;
